src/sentry/integrations/utils/metrics.py

```diff
@@ -173,6 +173,23 @@ def record_failure(
             self._extra.update(extra)
         self._terminate(EventLifecycleOutcome.FAILURE, exc)
 
+    def record_halt(self, exc: BaseException | None = None) -> None:
```

Comment:
I think unfortunately we have more cases like this. Can you share this broadly with the team so they can reuse it in their integration code?